python/cudf/cudf/core/subword_tokenizer.py

```diff
@@ -0,0 +1,279 @@
+                "Currently padding to the provided max_length is supported"
+            )
+            raise NotImplementedError(error_msg)
+
+        if max_length <= stride:
+            error_msg = "Stride should be less than max_length"
+            raise ValueError(error_msg)
+
+        if return_tensors not in ["cp", "pt", "tf"]:
+            error_msg = (
+                "Only cupy(cp), pytorch(pt) and tensorflow(tf)"
+                "tensors are supported"
+            )
+            raise NotImplementedError(error_msg)
+
+        stride = max_length - stride
+        # behaviour varies from subword_tokenize but maps with huggingface
+
+        input_ids, attention_mask, metadata = cpp_subword_tokenize(
+            text._column,
+            self.vocab_file,
+            max_sequence_length=max_length,
+            stride=stride,
+            do_lower=self.do_lower_case,
+            do_truncate=truncation,
+            max_rows_tensor=max_num_rows,
+        )
+
+        tokenizer_output = dict()
+        tokenizer_output["input_ids"] = cp.asarray(input_ids).reshape(
+            -1, max_length
+        )
+        tokenizer_output["attention_mask"] = cp.asarray(
+            attention_mask
+        ).reshape(-1, max_length)
+        tokenizer_output["metadata"] = cp.asarray(metadata).reshape(-1, 3)
+
+        if add_special_tokens:
+            tokenizer_output = _bert_add_special_tokens(tokenizer_output)
+
+        tokenizer_output = {
+            k: _cast_to_appripate_type(v, return_tensors)
+            for k, v in tokenizer_output.items()
+        }
+
+        return tokenizer_output
+
+
+def _bert_add_special_tokens(token_o):
+
+    max_length = token_o["input_ids"].shape[1]
+    seq_end_col = max_length - (token_o["input_ids"][:, ::-1] != 0).argmax(1)
+    # clipping to take overflow into account
+    seq_end_col = cp.clip(seq_end_col + 1, a_max=max_length - 1)
+
+    _bert_add_special_tokens_input_ids(token_o["input_ids"], seq_end_col)
+    _bert_add_special_tokens_attention_mask(
+        token_o["attention_mask"], seq_end_col
+    )
+    _bert_add_spedical_tokens_metadata(token_o["metadata"], max_length)
+
+    return token_o
+
+
+def _bert_add_special_tokens_input_ids(input_ids, seq_end_col):
+    # Mark sequence start with [CLS] token mapping to the start of sequence
+    input_ids[:, 1:-1] = input_ids[:, 0:-2]
+    input_ids[:, 0] = 101
+    # Mark end of sequence [SEP]
+
+    input_ids[
+        cp.arange(0, input_ids.shape[0], dtype=cp.uint32), seq_end_col
+    ] = 102
+
+
+def _bert_add_special_tokens_attention_mask(attention_mask, seq_end_col):
+    # Copy attention masks for all but last two
+    attention_mask[:, 1:-1] = attention_mask[:, 0:-2]
+    # Mark [CLS] token with 1
+    attention_mask[:, 0] = 1
+    # Mark [SEP] token with 1
+    attention_mask[
+        cp.arange(0, attention_mask.shape[0], dtype=cp.uint32), seq_end_col
+    ] = 1
+
+
+def _bert_add_spedical_tokens_metadata(metadata, max_length):
```

Comment:
Should this be special instead of spedical?

Also, add comment or brief docstring

Reply:
Resolved with commit https://github.com/rapidsai/cudf/commit/0018779bde5984931e114a3cd412ce869706a7b3